Assets/MixedRealityToolkit.Services/InputSimulation/KeyBinding.cs

```diff
@@ -0,0 +1,255 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using UnityEngine;
+using System;
+using System.Collections.Generic;
+using System.Runtime.CompilerServices;
+
+[assembly: InternalsVisibleTo("Microsoft.MixedReality.Toolkit.Services.InputSimulation.Editor")]
+namespace Microsoft.MixedReality.Toolkit.Input
+{
+    /// <summary>
+    /// Identifier of a key combination or mouse button for generic input binding.
+    /// </summary>
+    /// <remarks>
+    /// This encodes either a KeyCode with optional modifiers or a mouse button index.
+    /// </remarks>
+    [System.Serializable]
+    public struct KeyBinding
+    {
+        /// <summary>
+        /// The type of value encoded in the <see cref="code"/> property.
+        /// </summary>
+        public enum KeyType : int
+        {
+            None = 0,
+            Mouse = 1,
+            Key = 2,
+        }
+
+        /// <summary>
+        /// Enum for interpreting the mouse button integer index.
+        /// </summary>
+        public enum MouseButton : int
+        {
+            Left = 0,
```

Comment:
> Left = 0, [](start = 11, length = 10)

Nit: I think that our guidelines say to include a None value like:

[https://docs.microsoft.com/en-us/dotnet/standard/design-guidelines/enum](https://docs.microsoft.com/en-us/dotnet/standard/design-guidelines/enum)

I think we can go either way, it would be good to be consistent within the same file (i.e. KeyType.None and MouseButton.None). If only because this would also let us return MouseButton.None from the "TryGet..." function down below in failure case (instead of returning a "valid" mouse button)

Reply:
Ah... this was done to keep some consistency with the existing InputSimulationMouseButton, I guess nevermind then.

---
In reply to: [320505346](https://github.com/microsoft/MixedRealityToolkit-Unity/pull/5853#discussion_r320505346) [](ancestors = 320505346)